### PR TITLE
Add Item to Shop screen with search, filtering, and quantity picker

### LINKED ISSUE #14: App: Add Item to Shop screen
## Summary
Implement the screen for browsing the item catalog and adding items to a shop's inventory.

## Blocked by
- #5 Shop CRUD Use Cases
- #8 Repository Implementations
- #9 Metro DI, Navigation & Theme

### AddItemToShopViewModel
- Load all catalog items (built-in + custom) via `ItemRepository`
- Support search by name and filter by category
- `addItem(itemId, quantity)` — delegates to `AddItemToShopUseCase`
- Track which items have been added (for visual feedback)

### UI
- **Search bar** at top for filtering items by name
- **Category filter** chips (All, Weapons, Armor, Potions, etc.)
- **Item list**: each row shows name, category, base price, rarity badge
- **Tap item** to add to shop — show quantity picker (default 1, option for unlimited/∞)
- Visual indicator for already-added items
- **Navigation**: back to Shop Detail after adding

### Future Consideration
- "Create Custom Item" button/flow could be included here or deferred
- For V1, at minimum support adding existing catalog items

### ViewModel Tests
- Test catalog items load
- Test search filters items
- Test category filter works
- Test add item to shop succeeds

## Acceptance Criteria
- [ ] Catalog items display with search and category filtering
- [ ] Tapping an item adds it to the shop's inventory
- [ ] Quantity can be specified when adding
- [ ] Already-added items are visually indicated
- [ ] Navigation back to Shop Detail works
- [ ] ViewModel unit tests pass

### PR DESCRIPTION
**Pull request by mjl757**
## Summary
Implement the Add Item to Shop screen that allows GMs to browse the item catalog and add items to a shop's inventory with search, category filtering, and quantity selection.

### :app module
- **AddItemToShopViewModel**: Loads all catalog items via `ItemRepository`, supports search by name and filter by category, tracks added items for visual feedback, delegates to `ShopRepository.addItemToShop`
- **AddItemToShopScreen**: Compose UI with search bar, horizontally scrollable category filter chips (All + all ItemCategory values), lazy item list showing name/category/price/rarity badge, tap-to-add with quantity picker dialog (supports specific quantity and unlimited stock), check icon for already-added items
- **AddItemToShopViewModelTest**: 11 unit tests covering catalog loading, search filtering, category filtering, combined filters, item addition (with quantity and unlimited), nonexistent item handling, and multiple item tracking

### Build configuration
- Added `compose-material-icons-extended` dependency for Material icons
- Added `androidx-lifecycle-viewmodel-compose` dependency for ViewModel support
- Added `turbine` test dependency for Flow testing

## Testing
- 11 ViewModel unit tests written using JUnit 5 + Turbine
- Tests cover: initial loading state, catalog load, search by name (case-insensitive), category filter, clearing filters, combined search+category, add item success, unlimited quantity, nonexistent item, empty search reset, multiple adds
- [x] All tests pass (`./gradlew :app:testDebugUnitTest`)
- [x] Build succeeds (`./gradlew build`)

Closes #14

### CODE REVIEW

File: app/src/main/kotlin/com/shopforge/ui/additem/AddItemToShopScreen.kt [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+    isAdded: Boolean,
+    onTap: () -> Unit,
+) {
+    ListItem(
+        modifier = Modifier.clickable(onClick = onTap),
+        headlineContent = {
+            Text(text = item.name)
+        },
+        supportingContent = {
+            Text(
```
Comment (mjl757) [inline app/src/main/kotlin/com/shopforge/ui/additem/AddItemToShopScreen.kt]: Nit: `selectedItemForAdd!!` force-unwrap on line 231 works because of the null check on line 228, but it's fragile if the condition is ever refactored. Consider binding to a local val at the top of the block:

```kotlin
val item = selectedItemForAdd ?: return
QuantityPickerDialog(
    itemName = item.name,
    ...
)
```

Or use the idiomatic `?.let` pattern.

**Review by mjl757 — COMMENTED** [review]
## PR Review: Add Item to Shop screen

Overall this is a well-structured PR. The ViewModel is clean, the Compose UI follows good patterns (hoisted state, event lambdas), and the test coverage is solid with 11 tests covering the key scenarios. A few issues worth addressing:

### Duplicate dependency in version catalog

**gradle/libs.versions.toml** -- The PR adds `compose-material-icons-extended` but the catalog already defines `compose-icons-extended` with the exact same Maven coordinate (`androidx.compose.material:material-icons-extended`). This means `app/build.gradle.kts` will end up with **three** icon dependencies where two are identical:
```
compose-material-icons-extended  ->  androidx.compose.material:material-icons-extended  (NEW, duplicate)
compose-icons-core               ->  androidx.compose.material:material-icons-core
compose-icons-extended           ->  androidx.compose.material:material-icons-extended  (EXISTING)
```
The new `compose.material.icons.extended` entry should be removed and the existing `compose.icons.extended` used instead. The `compose-icons-core` is also redundant since `material-icons-extended` transitively includes it, but that is a pre-existing issue.

### Silent failure when adding a nonexistent item

**AddItemToShopViewModel.kt, line 59** (`val item = itemRepository.getItemById(itemId) ?: return@launch`) -- If `getItemById` returns null, the function silently returns with no user feedback. While the test verifies this behavior, in practice this is a quiet failure. Consider logging a warning or setting an error message so the issue is at least observable during development.

### Race condition on `addedItemIds`

**AddItemToShopViewModel.kt, line 66** (`addedItemIds.value = addedItemIds.value + itemId`) -- If a user rapidly taps two different items, two coroutines could read the same `addedItemIds.value` before either writes, causing one add to be lost from the set. Use `update {}` instead:
```kotlin
addedItemIds.update { it + itemId }
```
`MutableStateFlow.update` uses a CAS loop to handle concurrent modifications safely.

### `selectedItemForAdd!!` force-unwrap

**AddItemToShopScreen.kt, line 232** -- The code checks `selectedItemForAdd != null` in the condition but then uses `selectedItemForAdd!!` on line 232. Since `selectedItemForAdd` is a `var` backed by mutable state, it could theoretically recompose to null between the check and the usage. A safer pattern would be:
```kotlin
selectedItemForAdd?.let { item ->
    QuantityPickerDialog(
        itemName = item.name,
        ...
    )
}
```
(The `onConfirm` lambda already uses `?.let` -- the inconsistency suggests this was partially addressed.)

### No upper bound on quantity

**AddItemToShopScreen.kt, QuantityPickerDialog** -- The increment button has no upper limit, so a user could tap `+` indefinitely. Consider capping at a reasonable maximum (e.g., 999) to prevent accidental or nonsensical values.

### Snackbar error could be dismissed without user seeing it

**AddItemToShopScreen.kt, lines 105-110** -- `onClearError()` is called immediately after `showSnackbar` returns (i.e., when the snackbar is dismissed). This is fine, but note that if `errorMessage` changes again while a snackbar is already showing, the `LaunchedEffect` will restart and the first snackbar gets canceled. This is standard Compose behavior but worth being aware of if errors can fire in quick succession.

### Extra blank line in Item.sq

**Item.sq** -- The diff adds a spurious blank line (line 65 in the new file). Very minor, but it creates an inconsistent double-blank between `countAll` and `lastInsertRowId`.

### Minor: `metrox-viewmodel-compose` already in dependencies

The PR adds `androidx-lifecycle-viewmodel-compose` but the project already has `metrox-viewmodel-compose` from the Metro DI library. Confirm these serve different purposes and both are needed -- if Metro's ViewModel Compose integration already provides `viewModel()` / `ViewModel` lifecycle support, the AndroidX dependency may be unnecessary.

### What looks good
- Clean separation between UI state, ViewModel, and screen composable
- Exhaustive `when` expressions on `ItemCategory` and `Rarity` ensure compile-time safety if new entries are added
- Good use of `SharingStarted.WhileSubscribed(5_000)` for lifecycle awareness
- Thorough test coverage including edge cases (nonexistent item, combined filters, unlimited quantity)
- Accessibility: semantic content descriptions on search field and icons

**Review by mjl757 — COMMENTED** [review]
## Review summary

The overall structure is solid — the screen is properly stateless, `collectAsStateWithLifecycle` is used correctly, the Turbine-based tests are well written, and the Koin `viewModel { (shopId: Long) -> ... }` binding with `parametersOf` is the right approach. A few correctness issues need attention before merging.

### Must fix

**1. ViewModel bypasses `AddItemToShopUseCase` (see inline comment on `AddItemToShopViewModel.kt:35`)**
`addItem()` calls `shopRepository.addItemToShop()` directly. `AddItemToShopUseCase` already exists, is registered in `domainModule`, and owns the only place where `require(quantity == null || quantity >= 0)` is enforced. This ViewModel should inject and delegate to `AddItemToShopUseCase` instead of `ShopRepository`, consistent with every other ViewModel in the project. Without this, passing `quantity = 0` silently creates a zero-stock inventory entry with no error or guard.

**2. `route.shopId.toLong()` can crash the composable (see inline comment on `ShopForgeNavHost.kt:97`)**
`AppRoute.AddItemToShop.shopId` is typed `String`. `.toLong()` throws `NumberFormatException` with no recovery path if the string is ever malformed. Use `.toLongOrNull()` with an explicit fallback, or change the route type to `Long`.

**3. `addedItemIds` does not reflect pre-existing inventory (see inline comment on `AddItemToShopViewModel.kt:97`)**
The set starts empty on every ViewModel creation, so items the shop already contains will not show the check mark, and tapping them silently increments their stock (per `addItemToShop`'s documented contract) rather than doing nothing. Seed the set from `shopRepository.getInventory(shopId)` on init.

### Should fix

**4. Snackbar / error-clear race (see inline comment on `AddItemToShopScreen.kt:73`)**
`LaunchedEffect(uiState.errorMessage)` cancels the in-progress `showSnackbar` call when a new error arrives, which means `onClearError()` from the cancelled coroutine clears the new error before it is ever displayed. Consider a one-shot event model (e.g. a `SharedFlow` channel or an event counter in the UI state).

**5. Dialog state lost on configuration change (see inline comment on `AddItemToShopScreen.kt:81`)**
`showQuantityDialog` and `selectedItemForAdd` are local `remember` state and do not survive rotation. Hoist the pending-item-to-add into `AddItemToShopUiState` so the dialog reappears correctly after a config change.

**6. Quantity-zero mismatch between dialog and use case (see inline comment on `AddItemToShopScreen.kt:304`)**
The dialog minimum is 1, but the use case guard allows 0 (`quantity >= 0`). This inconsistency is untested and undocumented. Decide on a canonical minimum and enforce it in one place.

### Test gap

**7. Missing validation test (see inline comment on `AddItemToShopViewModelTest.kt:87`)**
No test covers the invalid-quantity path. Once the ViewModel delegates to the use case, add a test that asserts `addItem(id, 0)` or `addItem(id, -1)` sets an error state instead of succeeding.
